packages/fxa-auth-server/lib/routes/linked-accounts.ts

```diff
@@ -174,29 +192,32 @@ export class LinkedAccountHandler {
         this.log.debug('handleGoogleSET.processing', {
           eventType,
         });
-        if (
-          googleEventHandlers[eventType as keyof typeof googleEventHandlers]
-        ) {
-          await googleEventHandlers[
-            eventType as keyof typeof googleEventHandlers
-          ](jwtPayload.events[eventType], this.log, this.db);
-          this.statsd.increment(`handleGoogleSET.processed.${eventType}`);
-          this.log.debug('handleGoogleSET.processed', {
-            eventType,
-          });
-        } else {
-          // Log that an unknown event type was received and ignore it
-          handleGoogleOtherEventType(eventType, this.log);
-          this.statsd.increment(
-            `handleGoogleSET.unknownEventType.${eventType}`
-          );
+
+        try {
+          if (
+            googleEventHandlers[eventType as keyof typeof googleEventHandlers]
+          ) {
+            await googleEventHandlers[
+              eventType as keyof typeof googleEventHandlers
+            ](jwtPayload.events[eventType], this.log, this.db, this.statsd);
+            this.statsd.increment(`handleGoogleSET.processed.${eventType}`);
+            this.log.debug('handleGoogleSET.processed', {
+              eventType,
+            });
+          } else {
+            // Log that an unknown event type was received and ignore it
+            handleGoogleOtherEventType(eventType, this.log);
+            this.statsd.increment(
+              `handleGoogleSET.unknownEventType.${eventType}`
+            );
+          }
+        } catch (eventError) {
+          this.statsd.increment(`handleGoogleSET.eventHandler.error.${eventType}`);
```

Comment:
Just double checking that we are pretty confident that there's not going to be ton of different values for `eventType`?

You could also do, `this.statsd.increment(`handleGoogleSET.eventHandler.error`, [eventType]);` which might making graphing easier.